src/Controllers/PoolProviderController.cs

```diff
@@ -140,7 +151,17 @@ public async Task<IActionResult> AcquireAgent([FromBody] AgentAcquireItem agentR
                     {"OrchestrationId", orchestrationId},
                     {"JobName", jobName},
                 };
-                var agentInfoItem = await jobCreator.CreateJob(cancellationToken, properties);
+
+                if (!string.IsNullOrEmpty(associatedJobInfo.SystemPullRequestTargetBranch))
+                {
+                    properties.Add("System.PullRequest.TargetBranch", associatedJobInfo.SystemPullRequestTargetBranch);
```

Comment:
This is like... a 3rd level hack.  I really thing this servicing queue stuff needs to be redesigned.  Faking a build property into the property bag that we generally aren't supposed to depend on so that we can choose a different queue without informing the user is... a lot.

Reply:
So what would be you recommendation here? Would extending the `JobCreationRequest` with additional `TargetBranch` property be a good solution? Or do you think that we shouldn't touch Pool Provider and Helix API at all and just do all redirection client-side? Or maybe something else entirely?